### DISCUSSION TITLE
How to Use Time Trigger

### DISCUSSION THREAD

**Discussion by 918483** [savonet/liquidsoap#1868] [Q&A]
I have a script in linux that generates the timetable speech every minute but when I send the liquidsoap script to read the mp3 it delays a thing of minutes as if it had loaded almost 15 minutes ago how to proceed I use liqudidsoap 2.0

**Comment by 918483** [comment]
Even when you order the file to be loaded, it's as if it had already loaded, then the time comes out with the wrong speech, is there a way for the file to be loaded in real time to be played at the correct time?

**Comment by 918483** [comment]
@toots ??

**Comment by toots** [comment]
Hi! I think we'd need more context about your script to be able to help:
* script, full or relevant extract
* logs, full or relevant extract
* what do you expect to see, what is happening?
* Liquidsoap version, installation method, OS version

**Comment by 918483** [comment]
Linux system, liquidsoap 2.0 installed via ssh command using .zip

In this case I hope to run the .mp3 audio in real time to hear the correct tempo in the audio that is generated by the script.
As I said, I have a script that every minute it generates the speech at the right time, but when liquidsoap looks for this .mp3 generated by the script it preloads so when it will play delayed speech at the wrong time because it was loaded way before current time to run.

Roteiro

clock = audio_to_stereo (amplify (0,5, override = "replay_gain", single (id="HoraCerta", "/home/streaming/horacerta/7062/hora_certa.mp3")))
radio =  rotate(weights = [1,5] ,[clock,radio])

Using this path above it loads the audio much earlier and as the script made outside of liquidsoap runs every minute it doesn't say the correct time when generating the type if now it's 19:58 and it's time to play the speech it would do say 19:50 because the .mp3 was loaded at this time, the correct thing would be for him to say 19:58 and not 19:50, which is when he read the generated .mp3

**Reply by 918483** [reply]
@toots ??

**Comment by 918483** [comment]
@toots Can you help me ?

**Comment by toots** [comment]
Hi,

What output do you use? `output.icecast` and potentially others have internal buffers that make it difficult to send audio with short latency.

More generally, low latency is tricky to obtain, you need to make sure that the whole chain to produce data, from liquidsoap's internal frame to the encoders to the outputs to the way your listeners are connected is properly set to minimize delays.
